arkouda/alignment.py

```diff
@@ -172,6 +175,63 @@ def in1dmulti(a, b, assume_unique=False, symmetric=False):
             return atruth
 
 
+def find(query, space):
+    """
+    Return indices of query items in a search list of items (-1 if not found).
+
+    Parameters
+    ----------
+    query : (sequence of) array-like
+        The items to search for. If multiple arrays, each "row" is an item.
+    space : (sequence of) array-like
+        The set of items in which to search. Must have same shape/dtype as query.
+
+    Returns
+    -------
+    indices : pdarray, int64
+        For each item in query, its index in space or -1 if not found.
+    """
+
+    # Concatenate the space and query in fast (block interleaved) mode
+    if isinstance(query, (pdarray, Strings, Categorical)):
+        if type(query) != type(space):
+            raise TypeError("Arguments must have same type")
+        c = concatenate((space, query), ordered=False)
+        spacesize = space.size
+        querysize = query.size
+    else:
+        atypes = np.array([ai.dtype for ai in query])
+        btypes = np.array([bi.dtype for bi in space])
+        if not (atypes == btypes).all():
+            raise TypeError("Array dtypes of arguments must match")
+        c = [concatenate((si, qi), ordered=False) for si, qi in zip(space, query)]
+        spacesize = space[0].size
+        querysize = query[0].size
+    # Combined index of space and query elements, in block interleaved order
+    # All space indices are less than all query indices
+    i = concatenate((arange(spacesize), arange(spacesize, spacesize + querysize)), ordered=False)
```

Comment:
I might be missing something but it's not immediately obvious to me how this is different than
```python
i = arange(spacesize + querysize)
```

Reply:
It ends up being different because of the `ordered=False`, which block-interleaves the two arrays, rather than sequentially concatenating them. The goal is to get a block-interleaved array of indices that mirrors the block-interleaved array of elements from `space` and `query`.

Reply:
ah okay! that makes sense